lib/Sema/CSSimplify.cpp

```diff
@@ -6966,6 +6971,13 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     return matchTupleTypes(matchingType, smaller, matchKind, subflags, locator);
   }
 
+  case FixKind::AllowAutoClosurePointerConversion: {
+    if (recordFix(fix))
+      return SolutionKind::Error;
+    return matchTypes(type1, type2, matchKind, subflags,
+        locator.withPathElement(ConstraintLocator::FunctionArgument));
```

Comment:
I'm not sure this is right – when matching types, `ConstraintLocator::FunctionArgument` is used to indicate that we're matching parameter types for a function conversion, which can add `SK_FunctionConversion` to the solution's score. This probably isn't a big deal for this case given we're going to be producing an error anyway, but perhaps we should be removing the `ConstraintLocator::AutoclosureResult` path element from the locator instead?

Reply:
I think that’s okay here, since the intent is to indicate that this is an argument, less trouble then removing last element and re-creating locator :)